src/server/pkg/storage/chunk/metadata.go

```diff
@@ -0,0 +1,89 @@
+package chunk
+
+import (
+	"context"
+	"crypto/sha512"
+	"encoding/hex"
+
+	"github.com/jmoiron/sqlx"
+)
+
+type ChunkID []byte
```

Comment:
Similar to what I mentioned below, I think we should lean on the package name for type namespacing when it makes sense. It is cleaner and idiomatic Go. So, I think this could just be `chunk.ID`.